x-hooks/use-octokit-graph.tsx

```diff
@@ -0,0 +1,293 @@
+
+    return graphql.defaults({
+      headers: {
+        authorization: `token ${user?.accessToken}`
+      }
+    });
+  }
+
+  /**
+   * Get all pages of a graphql list, the query MUST CONTAIN A `pageInfo`
+   * @param query string containing the GraphQL query
+   * @param variables that mus be passed to the GraphQL query
+   * @returns an array of objects containing the data returned by the GraphQL query
+   */
+  async function getAllPages(query, variables) {
+    const api = getOctoKitInstance();
+    
+    if (!api) return;
+
+    const pages = [];
+    let nextPageCursor = null;
+    let hasMorePages = false;
+
+    do {
+      const response = await api(query, {...variables, cursor: nextPageCursor});
+
+      pages.push(response);
+
+      const { endCursor, hasNextPage } = getPropertyRecursively("pageInfo", response);
+
+      nextPageCursor = endCursor;
+      hasMorePages = hasNextPage;
+
+    } while (hasMorePages);
+
+    return pages;
+  }
+
+  async function getPullRequestParticipants(repositoryPath:  string, pullId: number): Promise<string[]> {
+    const { owner, repo } = getOwnerRepoFrom(repositoryPath);
+
+    const response = await getAllPages(`
+        query PullRequestParticipants($repo: String!, $owner: String!, $pullId: Int!, $cursor: String) {
+          repository(name: $repo, owner: $owner) {
+              pullRequest(number: $pullId) {
+                  participants(first: 100, after: $cursor) {
+                      pageInfo {
+                          endCursor
+                          hasNextPage
+                      }
+                      nodes {
+                          login
+                      }
+                  }
+              }
+          }
+      }
+    `, {
+      repo,
+      owner,
+      pullId
+    });
+
+    const participants = 
+      response?.flatMap(item => getPropertyRecursively("nodes", item).map(node => node["login"]));
+
+    return participants;
+  }
+
+  async function getPullRequestLinesOfCode(repositoryPath:  string, pullId: number): Promise<number> {
+    const { owner, repo } = getOwnerRepoFrom(repositoryPath);
+
+    const githubAPI = getOctoKitInstance();
+
+    const response = await githubAPI?.(`
+        query PullRequestParticipants($repo: String!, $owner: String!, $pullId: Int!) {
+          repository(name: $repo, owner: $owner) {
+              pullRequest(number: $pullId) {
+                additions
+                deletions
+              }
+          }
+      }
+    `, {
+      repo,
+      owner,
+      pullId
+    });
+
+    const { additions, deletions } = response["repository"]["pullRequest"];
+
+    return additions + deletions;
+  }
+
+  async function getIssueOrPullRequestComments(repositoryPath:  string, id: number) {
+    const { owner, repo } = getOwnerRepoFrom(repositoryPath);
+
+    const response = await getAllPages(`
+        query Comments($repo: String!, $owner: String!, $id: Int!, $cursor: String) {
```

Comment:
put these queries into a `graph-ql-queries` folder, and then import them and use them